File: 1615-maximal-network-rank/1615-maximal-network-rank.js
/**
 * @param {number} n
 * @param {number[][]} roads
 * @return {number}
 */
var maximalNetworkRank = function(n, roads) {
  const connected = Array.from({ length: n }, () => new Array(n).fill(0));
  const cnt = new Array(n).fill(0);
  let ans = 0;
  
  for (let [a, b] of roads) {
    cnt[a]++;
    cnt[b]++;
    connected[a][b] = connected[b][a] = 1;
  }
  
  for (let i = 0; i < n; i++) {
    for (let j = i + 1; j < n; j++) {
      ans = Math.max(ans, cnt[i] + cnt[j] - connected[i][j]);
    }
  }
  
  return ans;
};

File: 1615-maximal-network-rank/1615-maximal-network-rank.py
class Solution:
    def maximalNetworkRank(self, n: int, roads: List[List[int]]) -> int:
        connected = [[0] * n for _ in range(n)]
        cnt = [0] * n
        ans = 0
        for a, b in roads:
            cnt[a] += 1
            cnt[b] += 1
            connected[a][b] = connected[b][a] = 1
        
        return max(cnt[i] + cnt[j] - connected[i][j] for i in range(n) for j in range(i+1, n))
        